.github/workflows/test-unit.yml

```diff
@@ -0,0 +1,52 @@
+name: test unit
+
+on:
+  push:
+    branches-ignore:
+      - main
+
+  # When Changesets opens a PR it does not trigger GitHub actions,
+  # because its token does not have permission to. This is a hack
+  # to allow one of us to trigger GitHub actions for a changesets PR
+  # by enabling automerge on the PR.
+  pull_request_target:
+    types:
+      - auto_merge_enabled
+    branches:
+      - main # the target branch of the PR
+    paths:
+      - "**/CHANGELOG.md" # only changesets releases touch changelogs
+
+jobs:
+  unit-tests:
+    if: github.head_ref != 'changeset-release/main'
+    name: "vitest"
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup
+      - name: Unit tests
```

Comment:
Optional: You can put the shard number in the step names :) We do this for the stories

Reply:
Might as well, since I need to start my branch again